app/experimenter/nimbus-ui/src/hooks/useExperiment.tsx

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { useQuery } from "@apollo/client";
+import { GET_EXPERIMENT_QUERY } from "../gql/experiments";
+import { getExperiment } from "../types/getExperiment";
+
+/**
+ * Hook to retrieve all Experiment data by slug.
+ *
+ * Result will be a loading state, a not found state, and the experiment typed
+ * as getExperiment['experimentBySlug']. Refer to generated type defs under
+ * ../types/getExperiment.ts
+ *
+ * Example:
+ *
+ * const experiment = useExperiment('the-slug');
+ *
+ * if (experiment === undefined) {
+ *   return <Loading />;
+ * }
+ *
+ * if (experiment === null) {
```

Comment:
Do these examples need to be updated since it looks like we can check `if (notFound)`?

Reply:
Oh, yes, thank you!